resin/models/device.py

```diff
@@ -687,3 +699,41 @@ def move(self, uuid, app_name):
             'device', 'PATCH', params=params, data=data,
             endpoint=self.settings.get('pine_endpoint')
         )
+
+    def get_status(self, uuid):
+        """
+        Get the status of a device.
+
+        Args:
+            uuid (str): device uuid.
+
+        Raises:
+            DeviceNotFound: if device couldn't be found.
+
+        Returns:
+            str: status of a device. List of available statuses: Idle, Configuring, Updating, Offline and Post Provisioning.
+
+        Examples:
+            >>> resin.models.device.get_status('8deb12a58e3b6d3920db1c2b6303d1ff32f23d5ab99781ce1dde6876e8d143')
+            'Offline'
+
+        """
+
+        device = self.get(uuid)
+        if device['provisioning_state'] == 'Post-Provisioning':
+            return DeviceStatus.POST_PROVISIONING
+
+        seen = device['last_seen_time'] and (datetime.strptime(device['last_seen_time'], "%Y-%m-%dT%H:%M:%S.%fZ")).year < 2013
+        if not device['is_online'] and seen:
```

Comment:
This must be ``if not device['is_online'] and not seen``, or you'll end up inferring that the device is in configuring state instead of simply offline all the time.